src/utils/api.js

```diff
@@ -112,6 +112,46 @@ export const formatActionsByTimeOfDay = (actionsByTimeOfDayObject) => {
   }));
 };
 
+// helper function used in getActionsByWeekday below
+const getActionWeekday = (action) => {
+  const dateKey = 'createdAt';
+  // createdAt should have the format "2020-12-31T23:59:59.999Z"
+  const date = new Date(action[dateKey]);
+  const weekday = date.getDay();
+  return weekday;
+};
+
+// Takes array of action objects and returns an object with {key: value} pairs of {weekday: #-of-actions}
+export const getActionsByWeekday = (actions) => {
+  const actionsByWeekdayMap = new Map(actions?.countBy(getActionWeekday));
+  for (let weekday = 0; weekday < 7; weekday += 1) {
+    if (!actionsByWeekdayMap.has(weekday)) {
+      actionsByWeekdayMap.set(weekday, 0);
+    }
+  }
```

Comment:
I don't get what you do here. You shouldn't use `new Map` from an immutable. Instead `countBy` gives you an immutable map https://immutable-js.com/docs/v4.3.0/Map/#Map()
From what I know it does the same as a javascript Map.

Reply:
If you need to get rid of immutable because of `filterActions` you can use `toJS()`.

Reply:
The for loop is to set the non-exist entries for the Map. Otherwise, if there is no actions on Monday, we won't show Monday in the chart. wdyt?